src/odemis/gui/cont/tabs.py

```diff
@@ -3543,6 +3554,10 @@ def _onAlignMode(self, mode):
         if mode != "fiber-align" and main.spec_sel:
             main.spec_sel.position.unsubscribe(self._onFiberPos)
 
+        if mode != "fiber-align":
```

Comment:
You need do to the same thing in case the tab is hidden, in `.Show(False)` (next to the `pauseStreams()` call).

Reply:
Do you mean here?
```
        if not self.IsShown():
            # Shouldn't happen, but for safety, double check
            logging.warning("Alignment mode changed while alignment tab not shown")
            # Unsubscribe from data of photo-detector 2
            if self._fbdet2:
                self._fbdet2.data.unsubscribe(self._on_fbdet2_data)
```

Reply:
No, I mean that you should add some extra code in `Sparc2AlignTab.Show()`.

Reply:
in line 4114?

Reply:
Yes, somewhere around there.

Reply:
done!